inst/qml/common/GlmInputComponent.qml

```diff
@@ -0,0 +1,210 @@
+//
+// Copyright (C) 2013-2018 University of Amsterdam
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public
+// License along with this program.  If not, see
+// <http://www.gnu.org/licenses/>.
+//
+
+import QtQuick			2.8
+import QtQuick.Layouts  1.3
+import JASP.Controls	1.0
+import JASP.Widgets		1.0
+import JASP				1.0
+
+// All Analysis forms must be built with the From QML item
+Group {
+
+	function updateWeightsTitle() {
+		if (family.currentText == "Binomial")
+		{title = qsTr("Total Number of Trials")}
+		else
+		{title = qsTr("Weights")}
+		return title
+	}
+
+	VariablesForm
+	{
+		preferredHeight: 350
+
+		AvailableVariablesList
+		{
+			name:				"allVariablesList"
+		}
+
+		AssignedVariablesList
+		{
+			name:				"dependent"
+			title:				qsTr("Dependent variable")
+			allowedColumns:		["scale","ordinal", "nominal", "nominalText"]
+			singleVariable:		true
+		}
+
+		AssignedVariablesList
+		{
+			name:				"covariates"
+			title:				qsTr("Covariates")
+			allowedColumns:		["ordinal", "scale"]
+		}
+
+		AssignedVariablesList
+		{
+			name:				"factors"
+			title:				qsTr("Factors")
+			allowedColumns:		["scale", "ordinal", "nominal", "nominalText"]
+		}
+
+		AssignedVariablesList
+		{
+			name:              "weights"
+			title:             updateWeightsTitle();
```

Comment:
this is not wrong, just wondering whether
```suggestion
			title:            family.currentIndex === "Binomial" ? qsTr("Total Number of Trials") : qsTr("Weights")
```
would suffice? Seems like an easier solution

Reply:
It didn't work; I think I'll just keep using the function as it is for now.